### ISSUE TITLE
Logging Inconsistent For LongNet

### ISSUE THREAD

**Issue by nprisbrey** [DRAGNLabs/301r_retnet#80]
We need to be consistently logging the same values for all types of models. For the LongNet architecture, for example, [we only log Validation loss](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L409-L417) whereas [we record Validation loss and perplexity for the Transformer models](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L254-L272) and [RetNet models](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L103-L121).

Also, why are we only recording perplexity for Validation? I'd like to see loss and perplexity recorded for all models for all datasets.

**Comment by nprisbrey** [comment]
Also, why is [validation perplexity only logged `on_epoch`](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L118) and not [also `on_step` just like everything else](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L108)?

**Comment by DrewGalbraith** [comment]
Good points and easy fixes! I see no reason to have logging inconsistencies. I'll standardize these is an upcoming PR.

**Comment by DrewGalbraith** [comment]
I'll add a related issue. Look at the screenshots below:

**No LongNet on this Graph**
<img width="425" alt="image" src="https://github.com/user-attachments/assets/59e72dc2-ee0c-4f06-b30a-1374a422d1c0" />
**Only LongNet on this Graph**
<img width="423" alt="image" src="https://github.com/user-attachments/assets/c7f75634-c2be-45a2-b76a-b8158591e4b7" />
**Only LongNet on this Graph**
<img width="420" alt="image" src="https://github.com/user-attachments/assets/8a5b55c0-b1ae-4fe2-aed0-aebc6f2f7161" />

### Problem:
This is sort of mentioned by @nprisbrey. We log and only `on_epoch` in [RetNet's validation](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L103) and in [Transformer's validation](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L254). Tensorboard titles the resulting graph simply `val_loss`. However, because we log `on_epoch` _and_ `on_step` in the [LongNet's validation](https://github.com/DRAGNLabs/301r_retnet/blob/c934c1df2145f3c8aadf4a2264dba5653722a006/src/models.py#L409), TensorBoard creates two separate val_loss graphs. To distinguish them, it titles one graph `val_loss_epoch` and the other `val_loss_step`. Because it logs the LongNet losses under different logging titles, they won't appear on the same graph as RetNet's and Transformer's logs, which, as we stated earlier, are logged under 'val_loss'.

### Solution:
Lets set all logging for `train` and `validate` to `True` for `on_epoch` and `on_step`. I don't think it slows us down too much and more data never hurts. This is the case for `train_loss_epoch` and `train_loss_step`, for example, on which all three architectures have both set to `True` and all appear on the same tensorboard graph:

**All architectures appear on this graph**<img width="796" alt="image" src="https://github.com/user-attachments/assets/840322f2-0507-4744-a4f8-40f4f21815f1" />

**Comment by DrewGalbraith** [comment]
You'll also note in the above graphs that LongNet seems totally unaffected by `RMS Norm` vs  `Layer Norm`. The same is true for Transformer. It can be hard to tell becaused the lines perfectly overlap one another in the screenshot. For RetNet, however, the `val_loss` is much improved by using `layer norm` over `RMS norm`. The [RetNet paper](https://arxiv.org/pdf/2307.08621) says a lot about normalization, unlike the LongNet paper. At a glance, it seems ok to replace `RMS` with `layer`.

**Comment by DrewGalbraith** [comment]
I'll add that Lightning uses Tensorflow's tensorboard which handles naming as shown in their `EventFileWriter` [class](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/summary/writer/event_file_writer.py#L73-L84). This may be helpful as we look into future logging bugs.
Answer sourced [here](https://stackoverflow.com/a/79355879/21747932).
